control/smart_mpc_trajectory_follower/README.md

```diff
@@ -0,0 +1,328 @@
+<p align="center">
+  <a href="https://proxima-ai-tech.com/">
+    <img width="500px" src="./images/proxima_logo.png">
+  </a>
+</p>
+
+# Smart MPC Trajectory Follower
+
+Smart MPC (Model Predictive Control) is a control algorithm that combines model predictive control and machine learning. While inheriting the advantages of model predictive control, it solves its disadvantage of modeling difficulty with a data-driven method using machine learning.
+
+This technology makes it relatively easy to operate model predictive control, which is expensive to implement, as long as an environment for collecting data can be prepared.
+
+<p align="center">
+  <a href="https://youtu.be/j7bgK8m4-zg?si=p3ipJQy_p-5AJHOP)">
+    <image width="700px" src="./images/autoware_smart_mpc.png">
+  </a>
+</p>
+
+## Setup
+
+After building autoware, move to `control/smart_mpc_trajectory_follower` and run the following command:
+
+```bash
+pip3 install .
+```
+
+If you have already installed and want to update the package, run the following command instead:
+
+```bash
+pip3 install -U .
+```
+
+## Provided features
+
+### Trajectory following control based on iLQR/MPPI
+
+The control mode can be selected from "ilqr", "mppi", or "mppi_ilqr", and can be set as `mpc_parameter:system:mode` in [mpc_param.yaml](./smart_mpc_trajectory_follower/param/mpc_param.yaml).
+In "mppi_ilqr" mode, the initial value of iLQR is given by the MPPI solution.
+
+> [!NOTE]
+> With the default settings, the "mppi" mode does not work well due to insufficient number of samples. Due to the limitation of using only the CPU, it is difficult to increase the number of samples sufficiently.
+
+To perform a simulation, run the following command:
+
+```bash
+ros2 launch autoware_launch planning_simulator.launch.xml map_path:=$HOME/autoware_map/sample-map-planning vehicle_model:=sample_vehicle sensor_model:=sample_sensor_kit trajectory_follower_mode:=smart_mpc_trajectory_follower
+```
```

Comment:
```suggestion
To perform the smart_mpc controller in the planning simulation, run the following command:

```bash
ros2 launch autoware_launch planning_simulator.launch.xml map_path:=$HOME/autoware_map/sample-map-planning vehicle_model:=sample_vehicle sensor_model:=sample_sensor_kit trajectory_follower_mode:=smart_mpc_trajectory_follower
```

The control mode can be selected from "ilqr", "mppi", or "mppi_ilqr", and can be set as `mpc_parameter:system:mode` in [mpc_param.yaml](./smart_mpc_trajectory_follower/param/mpc_param.yaml).
In "mppi_ilqr" mode, the initial value of iLQR is given by the MPPI solution.

> [!NOTE]
> With the default settings, the performance of "mppi" mode is limited due to an insufficient number of samples. This issue is being addressed with ongoing work to introduce GPU support.
```

Reply:
.... suggestion seems not working well. Replase this part with 

---
To perform the smart_mpc controller in the planning simulation, run the following command:

```bash
ros2 launch autoware_launch planning_simulator.launch.xml map_path:=$HOME/autoware_map/sample-map-planning vehicle_model:=sample_vehicle sensor_model:=sample_sensor_kit trajectory_follower_mode:=smart_mpc_trajectory_follower
```

The control mode can be selected from "ilqr", "mppi", or "mppi_ilqr", and can be set as `mpc_parameter:system:mode` in [mpc_param.yaml](./smart_mpc_trajectory_follower/param/mpc_param.yaml).
In "mppi_ilqr" mode, the initial value of iLQR is given by the MPPI solution.

> [!NOTE]
> With the default settings, the performance of "mppi" mode is limited due to an insufficient number of samples. This issue is being addressed with ongoing work to introduce GPU support.
---

Reply:
Revised Note as suggested.
https://github.com/autowarefoundation/autoware.universe/pull/6805/commits/980abddb3e097ca16fddcd56a740693d9e8c88d4#diff-1b10418540e411525cff15236aca92b0e6b3f4f03ddfc58d723fb6980708f700R43